src/rgw/rgw_lua_request.cc

```diff
@@ -173,14 +173,19 @@ struct BucketMetaTable : public EmptyMetaTable {
   using Type = rgw::sal::Bucket;
 
   static int IndexClosure(lua_State* L) {
-    const auto bucket = reinterpret_cast<Type*>(lua_touserdata(L, lua_upvalueindex(1)));
+    const auto s = reinterpret_cast<req_state*>(lua_touserdata(L, lua_upvalueindex(1)));
+    const auto bucket = s->bucket.get();
```

Comment:
you're right, thanks.  forgot to fix that.
I changed it to push nil if the bucket is empty and the index is not  "Name", think that should be enough since at that point we do not have much info on the bucket, just the string from the URL.

Reply:
before that change, we were pushing nil instead of Bucket MetaTable at preRequest, so it behaves the same.

Reply:
If the bucket's name is the only relevant field for `preReq`, perhaps it's worth changing the current behavior to not support accessing other fields for empty/null buckets. (instead of crashing)

Reply:
@yuvalif 
In this case, the other fields of `BucketMetaTable` will have nil values. Would it be better to change documentation that those fields are now optional?

Reply:
agree.
we should also change the documentation that `Request.Bucket` is now not optional